scripts/zones/Aht_Urhgan_Whitegate/npcs/Iruki-Waraki.lua

```diff
@@ -16,6 +16,20 @@ require("scripts/globals/pets");
 
 
 function onTrade(player,npc,trade)
+
+    OperationTeatimeProgress = player:getVar("OperationTeatimeProgress");
+    
+    if (player:getQuestStatus(AHT_URHGAN,NO_STRINGS_ATTACHED) == QUEST_COMPLETED and player:hasItem(17859) == false) then
+        if (trade:getGil() == 10000 and trade:getItemCount() == 1 and player:getFreeSlotsCount() >= 1) then
+            player:addItem(17859);
+            player:messageSpecial(ITEM_OBTAINED,17859); -- animator
+        end
+    elseif (OperationTeatimeProgress == 1) then
+        if (trade:hasItemQty(4161,1) and trade:hasItemQty(5570,1)) then -- Trade Chai and Sleeping Potion
```

Comment:
this trade will eat any additional items player includes in trade window.

recommend to require globals/npc_util and see the functions there.  for this i'd use

npcUtil.tradeHas(trade, {4161, 5570})

and

player:confirmTrade() instead of player:tradeComplete().  Would also move the confirmTrade() into the onEventFinish for event 780.